src/portfft/committed_descriptor_impl.hpp

```diff
@@ -322,7 +316,7 @@ class committed_descriptor_impl {
    * vector of kernel ids, factors
    */
   template <Idx SubgroupSize>
-  std::tuple<detail::level, kernel_ids_and_metadata_t> prepare_implementation(std::size_t kernel_num) {
+  std::tuple<detail::level, std::size_t, kernel_ids_and_metadata_t> prepare_implementation(std::size_t kernel_num) {
```

Comment:
The tuple's new addtion is undocumented. I'd still love this to be its own class.